src/sql/platform/connection/common/connectionStatusManager.ts

```diff
@@ -49,6 +50,21 @@ export class ConnectionStatusManager {
 		return !!this.findConnection(id);
 	}
 
+	public changeConnectionUri(newUri: string, oldUri: string) {
+		let info = this.findConnection(oldUri);
+		if (!info) {
+			this._logService.error(`No connection found associated with old URI : '${oldUri}'`);
+			throw new Error(nls.localize('connectionStatusManager.noConnectionForUri', 'Could not find connection with uri: {0}', oldUri));
+		}
+		if (this._connections[newUri]) {
```

Comment:
Is this also going to cause problems like the other one if we're saving over an existing file?

Reply:
an existing file that is already opened in ADS

Reply:
If its caught, the thing is designed to return the original input, and nothing is changed here.